File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/model/body/ride/StartRideBody.java
package com.lattis.ellipse.data.network.model.body.ride;

import com.google.gson.annotations.SerializedName;
import com.lattis.ellipse.domain.model.Location;

/**
 * Created by ssd3 on 3/23/17.
 */

public class StartRideBody {



        @SerializedName("bike_id")
        private int bike_id;
        @SerializedName("latitude")
        private double latitude;
    @SerializedName("longitude")
    private double longitude;

        public StartRideBody(int bike_id, Location location) {
            this.bike_id = bike_id;
            this.latitude = location.getLatitude();
            this.longitude =location.getLongitude();
        }

}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/entity/body/ride/StartRideBody.kt
package com.lattis.data.entity.body.ride

import com.google.gson.annotations.SerializedName
import com.lattis.data.entity.body.v2.BikeHubPortBody
import com.lattis.domain.models.Location

class StartRideBody(
    @field:SerializedName("bike_id") private val bike_id: Int,
    location: Location,
    device_token:String
) {

    @SerializedName("latitude")
    private val latitude: Double

    @SerializedName("longitude")
    private val longitude: Double

    @SerializedName("device_token")
    private val device_token: String

    init {
        latitude = location.latitude
        longitude = location.longitude
        this.device_token = device_token
    }
}